### PR TITLE
New Resource: `azurerm_traffic_manager_azure_endpoint`, `azurerm_traffic_manager_external_endpoint`, and `azurerm_traffic_manager_nested_endpoint`

### LINKED ISSUE #9734: Split `azurerm_traffic_manager_endpoint`
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

At this point in time the `azurerm_traffic_manager_endpoint` is shoe-horning 3 different resources (Azure Endpoints, External Endpoints and Nested Endpoints) into a single resource - we should look to split this for 3.0.

### New or Affected Resource(s)

* `azurerm_traffic_manager_endpoint`

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #9724 noted this and adds a TODO to implement this for 3.0

### PR DESCRIPTION
**Pull request by mbfrahry**
Fixes #9734 

image

### CODE REVIEW

File: internal/services/trafficmanager/traffic_manager_nested_endpoint_resource.go [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+		Create: resourceArmTrafficManagerNestedEndpointCreateUpdate,
+		Read:   resourceArmTrafficManagerNestedEndpointRead,
+		Update: resourceArmTrafficManagerNestedEndpointCreateUpdate,
+		Delete: resourceArmTrafficManagerNestedEndpointDelete,
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := endpoints.ParseEndpointTypeID(id)
+			return err
+		}),
+
+		DeprecationMessage: features.DeprecatedInThreePointOh("The resource 'azurerm_traffic_manager_endpoint' has been deprecated in favor of 'azurerm_traffic_manager_azure_endpoint', 'azurerm_traffic_manager_external_endpoint', and 'azurerm_traffic_manager_nested_endpoint'."),
```
Comment (tombuildsstuff) [inline internal/services/trafficmanager/traffic_manager_nested_endpoint_resource.go:34]: This should be in the `azurerm_traffic_manager_endpoint` resource?

**Review by tombuildsstuff — APPROVED** [review]
hey @mbfrahry 

Thanks for this PR - I've taken a look through and left some comments inline, since they apply to all of these resources rather than comment every time I hope you don't mind but I've gone ahead and pushed a few commits to fix the pending comments. As such I'm going to ask @jackofallops to take another look - but I think this should now be good to go 👍 

Thanks!

**Comment by tombuildsstuff** [comment]
Tests pass:

```
 $ TF_ACC=1 go test -v ./internal/services/trafficmanager -run="(TestAccAzureEndpoint_|TestAccExternalEndpoint_|TestAccNestedEndpoint_)" -timeout=60m
=== RUN   TestAccAzureEndpoint_basic
=== PAUSE TestAccAzureEndpoint_basic
=== RUN   TestAccAzureEndpoint_requiresImport
=== PAUSE TestAccAzureEndpoint_requiresImport
=== RUN   TestAccAzureEndpoint_complete
=== PAUSE TestAccAzureEndpoint_complete
=== RUN   TestAccAzureEndpoint_subnets
=== PAUSE TestAccAzureEndpoint_subnets
=== RUN   TestAccExternalEndpoint_basic
=== PAUSE TestAccExternalEndpoint_basic
=== RUN   TestAccExternalEndpoint_requiresImport
=== PAUSE TestAccExternalEndpoint_requiresImport
=== RUN   TestAccExternalEndpoint_complete
=== PAUSE TestAccExternalEndpoint_complete
=== RUN   TestAccExternalEndpoint_subnets
=== PAUSE TestAccExternalEndpoint_subnets
=== RUN   TestAccNestedEndpoint_basic
=== PAUSE TestAccNestedEndpoint_basic
=== RUN   TestAccNestedEndpoint_requiresImport
=== PAUSE TestAccNestedEndpoint_requiresImport
=== RUN   TestAccNestedEndpoint_complete
=== PAUSE TestAccNestedEndpoint_complete
=== RUN   TestAccNestedEndpoint_subnet
=== PAUSE TestAccNestedEndpoint_subnet
=== CONT  TestAccAzureEndpoint_basic
=== CONT  TestAccExternalEndpoint_complete
=== CONT  TestAccAzureEndpoint_subnets
=== CONT  TestAccNestedEndpoint_requiresImport
=== CONT  TestAccNestedEndpoint_subnet
=== CONT  TestAccNestedEndpoint_complete
=== CONT  TestAccExternalEndpoint_requiresImport
=== CONT  TestAccNestedEndpoint_basic
=== CONT  TestAccExternalEndpoint_subnets
=== CONT  TestAccAzureEndpoint_complete
--- PASS: TestAccExternalEndpoint_subnets (106.78s)
=== CONT  TestAccAzureEndpoint_requiresImport
--- PASS: TestAccNestedEndpoint_subnet (108.46s)
=== CONT  TestAccExternalEndpoint_basic
--- PASS: TestAccNestedEndpoint_basic (118.13s)
--- PASS: TestAccNestedEndpoint_requiresImport (120.23s)
--- PASS: TestAccExternalEndpoint_requiresImport (129.98s)
--- PASS: TestAccAzureEndpoint_basic (134.39s)
--- PASS: TestAccAzureEndpoint_subnets (146.00s)
--- PASS: TestAccExternalEndpoint_basic (108.26s)
--- PASS: TestAccExternalEndpoint_complete (234.15s)
--- PASS: TestAccNestedEndpoint_complete (242.75s)
--- PASS: TestAccAzureEndpoint_complete (254.23s)
--- PASS: TestAccAzureEndpoint_requiresImport (229.35s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/trafficmanager	338.177s
```

File: internal/services/trafficmanager/nested_endpoint_resource.go [hunk 2]
```diff
@@ -0,0 +1,340 @@
+package trafficmanager
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/trafficmanager/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/trafficmanager/validate"
+
```
Comment (jackofallops) [inline internal/services/trafficmanager/nested_endpoint_resource.go:10]: remove and sort?
```suggestion
```

File: website/docs/r/traffic_manager_endpoint.html.markdown [hunk 3]
```diff
@@ -10,6 +10,8 @@ description: |-
 
 Manages a Traffic Manager Endpoint.
 
+~> **NOTE:** This resource is **deprecated** in favour of the `azurerm_traffic_manager_azure_endpoint`, `azurerm_traffic_manager_external_endpoint`, or `azurerm_traffic_manager_nested_endpoint` resources and will be removed in version 3.0 of the Azure Provider.
```
Comment (jackofallops) [inline website/docs/r/traffic_manager_endpoint.html.markdown:13]: The resource won't flag itself as deprecated until the beta flag is active, should the phrasing on this note reflect it is _going_ to be deprecated?

**Review by jackofallops — APPROVED** [review]
Thanks @mbfrahry & @tombuildsstuff 
A few minor notes / questions in-line below, but otherwise LGTM 👍

File: website/docs/r/traffic_manager_endpoint.html.markdown (hunk 3 above)
Comment (tombuildsstuff) [reply website/docs/r/traffic_manager_endpoint.html.markdown:13]: Fixed by deprecating immediately 👍

**Pull request merged**